nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageServiceImpl.java

```diff
@@ -422,6 +427,34 @@ public ColdStorageContentStatus checkColdStorageContentAvailability(CoreSession
         return new ColdStorageContentStatus(beingRetrieved, available);
     }
 
+    @Override
+    public ColdStorageClassStatus checkColdStorageClass(CoreSession session) {
+
+        log.debug("Start checking the class storage for repository: {}", session::getRepositoryName);
+
+        DocumentModelList documents = session.query(GET_COLDSTORAGE_DOCUMENTS_TO_CHECK_QUERY);
+        int totalToUpdate = documents.size();
+        int nbDocumentsUpdated = 0;
+        for (DocumentModel documentModel : documents) {
+            BlobStatus blobStatus = getBlobStatus(documentModel);
+            if (blobStatus == null) {
+                continue;
+            }
+
+            if (!blobStatus.isDownloadable()) {
+                totalToUpdate--;
+                nbDocumentsUpdated++;
+                documentModel.setPropertyValue(COLD_STORAGE_CONTENT_STORAGE_CLASS_TO_UPDATED, false);
+                documentModel.setPropertyValue(COLD_STORAGE_CONTENT_AVAILABLE_IN_COLDSTORAGE, true);
+                session.saveDocument(documentModel);
+            }
+        }
```

Comment:
You should probably use a paginated query here in order to not stress the DB too much.

You can find an example [here](https://github.com/nuxeo/nuxeo-lts/blob/f65f044d73bfc9b7426d38457107e8b4619af6d5/modules/platform/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/listener/CheckedInCommentListener.java#L71)